### PR TITLE
Add setting to change first day of the week: Issue #844

### PR DESCRIPTION
**Pull request by biohzrddd**
Resolves issue #844 .

### CODE REVIEW

**Comment by biohzrddd** [comment]
[Screencast from 16-04-23 13:58:30.webm](https://user-images.githubusercontent.com/10577752/232326014-75a596e1-1ed6-4543-84c9-46f1a76c4c76.webm)

**Comment by j-f1** [comment]
I’m not fully on board with adding a new setting for this (I’d rather just pull from the user’s current locale) but I’m also not super strongly opposed to adding it. What do you think, @MatissJanis?

**Comment by MatissJanis** [comment]
We can't really pull it from the locale though. Most Europeans consider Monday to be the first day of the week. If we pull the first day of the week from users locale - what happens with Europeans in the US?

Adding it as a new setting is the only other solution I can think of.. so I'm OK with the proposal.

**Comment by biohzrddd** [comment]
@MatissJanis My thoughts exactly. Until we can figure out an automagical solution, this lets the non-US people be a bit more comfortable with the app and more likely to use it.

File: packages/desktop-client/src/components/settings/Format.js [hunk 1]
```diff
@@ -98,6 +115,17 @@ export default function FormatSettings({ prefs, savePrefs }) {
               />
             </Button>
           </Column>
+
+          <Column title="First day of the week">
+            <Button bounce={false} style={{ padding: 0 }}>
+              <CustomSelect
+                value={firstDayOfWeekIdx}
+                onChange={onFirstDayOfWeek}
+                options={daysOfWeek.map(f => [f.value, f.label])}
```
Comment (j-f1) [inline packages/desktop-client/src/components/settings/Format.js:124]: An alternative, which would make it easier to localize in the future (although 

```js
import locale from 'date-fns/locale/en-US';
```

```suggestion
                options={Array(7).fill().map((_, i) => [String(i), locale.localize.day(i)])}
```

Or:

```suggestion
                options={daysOfWeek.map((day, i) => [String(i), day])}
```

and then:

```js
let daysOfWeek = [
  'Sunday',
  'Monday',
  'Tuesday',
  'Wednesday',
  'Thursday',
  'Friday',
  'Saturday',
];
```

File: packages/desktop-client/src/components/settings/Format.js [hunk 2]
```diff
@@ -35,6 +47,10 @@ function Column({ title, children }) {
 }
 
 export default function FormatSettings({ prefs, savePrefs }) {
+  function onFirstDayOfWeek(idx) {
+    savePrefs({ firstDayOfWeekIdx: idx });
```
Comment (j-f1) [inline packages/desktop-client/src/components/settings/Format.js:51]:
```suggestion
    savePrefs({ firstDayOfWeekIdx: parseInt(idx) });
```

File: packages/desktop-client/src/components/settings/Format.js [hunk 3]
```diff
@@ -98,6 +115,17 @@ export default function FormatSettings({ prefs, savePrefs }) {
               />
             </Button>
           </Column>
+
+          <Column title="First day of the week">
+            <Button bounce={false} style={{ padding: 0 }}>
+              <CustomSelect
+                value={firstDayOfWeekIdx}
```
Comment (j-f1) [inline packages/desktop-client/src/components/settings/Format.js:122]:
```suggestion
                value={String(firstDayOfWeekIdx)}
```

File: packages/desktop-client/src/components/select/DateSelect.js [hunk 4]
```diff
@@ -107,6 +110,7 @@ export let DatePicker = React.forwardRef(
       picker.current = new Pikaday({
         theme: 'actual-date-picker',
         keyboardInput: false,
+        firstDay: stringToInteger(firstDayOfWeekIdx),
```
Comment (j-f1) [inline packages/desktop-client/src/components/select/DateSelect.js:113]:
```suggestion
        firstDay: firstDayOfWeekIdx,
```

**Review by j-f1 — COMMENTED** [review]
Suggestion: change the saved preference value to a number

File: packages/desktop-client/src/components/settings/Format.js [hunk 5]
```diff
@@ -8,6 +8,18 @@ import { Checkbox } from '../forms';
 
 import { Setting } from './UI';
 
+// Follows Pikaday 'firstDay' numbering
+// https://github.com/Pikaday/Pikaday
+let daysOfWeek = [
+  { value: '0', label: 'Sunday' },
```
Comment (biohzrddd) [inline packages/desktop-client/src/components/settings/Format.js:14]:
```suggestion
  'Sunday',
```

File: packages/desktop-client/src/components/settings/Format.js [hunk 6]
```diff
@@ -8,6 +8,18 @@ import { Checkbox } from '../forms';
...
+  { value: '1', label: 'Monday' },
```
Comment (biohzrddd) [inline packages/desktop-client/src/components/settings/Format.js:15]:
```suggestion
  'Monday',
```

File: packages/desktop-client/src/components/settings/Format.js [hunk 7]
```diff
@@ -8,6 +8,18 @@ import { Checkbox } from '../forms';
...
+  { value: '2', label: 'Tuesday' },
```
Comment (biohzrddd) [inline packages/desktop-client/src/components/settings/Format.js:16]:
```suggestion
  'Tuesday',
```

File: packages/desktop-client/src/components/settings/Format.js [hunk 8]
```diff
@@ -8,6 +8,18 @@ import { Checkbox } from '../forms';
...
+  { value: '3', label: 'Wednesday' },
```
Comment (biohzrddd) [inline packages/desktop-client/src/components/settings/Format.js:17]:
```suggestion
  'Wednesday',
```

File: packages/desktop-client/src/components/settings/Format.js [hunk 9]
```diff
@@ -8,6 +8,18 @@ import { Checkbox } from '../forms';
...
+  { value: '4', label: 'Thursday' },
```
Comment (biohzrddd) [inline packages/desktop-client/src/components/settings/Format.js:18]:
```suggestion
  'Thursday',
```

File: packages/desktop-client/src/components/settings/Format.js [hunk 10]
```diff
@@ -8,6 +8,18 @@ import { Checkbox } from '../forms';
...
+  { value: '5', label: 'Friday' },
```
Comment (biohzrddd) [inline packages/desktop-client/src/components/settings/Format.js:19]:
```suggestion
  'Friday',
```

File: packages/desktop-client/src/components/settings/Format.js [hunk 11]
```diff
@@ -8,6 +8,18 @@ import { Checkbox } from '../forms';
...
+  { value: '6', label: 'Saturday' },
```
Comment (biohzrddd) [inline packages/desktop-client/src/components/settings/Format.js:20]:
```suggestion
  'Saturday',
```

File: packages/desktop-client/src/components/select/DateSelect.js [hunk 12]
```diff
@@ -189,6 +193,12 @@ export default function DateSelect({
   let [selectedValue, setSelectedValue] = useState(value);
   let userSelectedValue = useRef(selectedValue);
 
+  const firstDayOfWeekIdx = useSelector(state =>
+    state.prefs.local.hasOwnProperty('firstDayOfWeekIdx')
+      ? state.prefs.local.firstDayOfWeekIdx
+      : '0',
```
Comment (biohzrddd) [inline packages/desktop-client/src/components/select/DateSelect.js:199]:
```suggestion
      : 0,
```

File: packages/desktop-client/src/components/settings/Format.js [hunk 13]
```diff
@@ -48,6 +64,7 @@ export default function FormatSettings({ prefs, savePrefs }) {
     savePrefs({ hideFraction });
   }
 
+  let firstDayOfWeekIdx = prefs.firstDayOfWeekIdx || '0'; // Sunday
```
Comment (biohzrddd) [inline packages/desktop-client/src/components/settings/Format.js:67]:
```suggestion
  let firstDayOfWeekIdx = prefs.firstDayOfWeekIdx || 0; // Sunday
```

**Review by biohzrddd — COMMENTED** [review]
Integrating changes from @j-f1, using int storage rather than char. We can add the localization later.

**Review by MatissJanis — APPROVED** [review]
Amazing!

**Review by j-f1 — APPROVED** [review]
Thanks!

**Pull request merged**